oauth2client/client.py

```diff
@@ -1222,6 +1227,31 @@ def create_scoped(self, scopes):
         """
         return self
 
+    @classmethod
+    def from_json(cls, s):
+        from oauth2client.service_account import _ServiceAccountCredentials
+        s = _from_bytes(s)
+        data = json.loads(s)
+
+        # A GoogleCredentials.from_stream return value might be a
```

Comment:
The goal of these methods is to make it so any object returned by "GoogleCredentials.from_stream" can be cached with from_json and to_json. Basically, this codepath should work:

creds = GoogleCredentials.from_stream(filename)
serialized_data = creds.to_json()
deserialized_object = GoogleCredentials.from_json(serialized_data)

since GoogleCredentials.from_stream returns _ServiceAccount instances, this method has to be able to deserialize _ServiceAccount instances.

Does that make sense?